Add details to Hardware section (#200)
wpilibsuite/frc-docs: source/docs/hardware/hardware-basics/preemptive-troubleshooting.rst

Radio Barrel Jack
-----------------

Make sure the correct barrel jack is used, not one that is too small and falls out for no reason.  This isn’t common, but ask an FTA and every once in awhile a team will use some random barrel jack that is not sized correctly, and it falls out in a match on first contact.

Ethernet Cable
--------------

If the RIO to radio ethernet cable is missing the clip that locks the connector in, get another cable.  This is a common problem that will happen several times in every competition. Make sure that your cables are secure. The clip often breaks off, especially when pulling it through a tight path, it snags on something then breaks.

Loose Cables
------------

Cables must be tightened down, particularly the radio power and ethernet cable.  The radio power cables don’t have a lot of friction force and will fall out (even if it is the correct barrel) if the weight of the cable-slack is allowed to swing freely.

Ethernet cable is also pretty heavy, if it’s allowed to swing freely, the plastic clip may not be enough to hold the ethernet pin connectors in circuit.

Reproducing Problems in the Pit
-------------------------------

> Beyond the normal shaking and rattling of all cables while the robot is power and tethered, you might try picking up one side of the robot  off the ground and drop it, and see if you lose connection. The driving on the field, especially when trying to breach defenses will often be very violent. It's better to see it fail in the pit rather than in a critical match.

Reviewer: ```suggestion
Beyond the normal shaking and rattling of all cables whilst the robot is power and tethered, you might try picking up one side of the robot off the ground and drop it, and see if you lose connection. Driving on the field, especially against defenders, will often be very violent. It's better to see it fail in the pit rather than in a critical match.
```

Radio Barrel Jack
-----------------

Make sure the correct barrel jack is used, not one that is too small and falls out for no reason.  This isn’t common, but ask an FTA and every once in awhile a team will use some random barrel jack that is not sized correctly, and it falls out in a match on first contact.

Ethernet Cable
--------------

If the RIO to radio ethernet cable is missing the clip that locks the connector in, get another cable.  This is a common problem that will happen several times in every competition. Make sure that your cables are secure. The clip often breaks off, especially when pulling it through a tight path, it snags on something then breaks.

Loose Cables
------------

Cables must be tightened down, particularly the radio power and ethernet cable.  The radio power cables don’t have a lot of friction force and will fall out (even if it is the correct barrel) if the weight of the cable-slack is allowed to swing freely.

Ethernet cable is also pretty heavy, if it’s allowed to swing freely, the plastic clip may not be enough to hold the ethernet pin connectors in circuit.

Reproducing Problems in the Pit
-------------------------------

> Beyond the normal shaking and rattling of all cables whilst the robot is power and tethered, you might try picking up one side of the robot off the ground and drop it, and see if you lose connection. Driving on the field, especially against defenders, will often be very violent. It's better to see it fail in the pit rather than in a critical match.

Reviewer: ```suggestion
Beyond the normal shaking of cables whilst the robot is powered and tethered, it is suggested that one side of the robot be picked up and dropped. Driving on the field, especially against defenders, will often be very violent, and this helps makes sure nothing falls out. It is better for the robot to fail in the pits rather than in the middle of a match.
```

Even this sounds slightly strange to me.

    In *FIRST*\ |reg| Robotics Competition, robots take a lot of stress while driving around the field. It is important to make sure that connections are tight, parts are bolted securely in place and that everything is mounted so that a robot bouncing around the field does not break.

> Check battery connections

Reviewer 2: ```suggestion
Check Battery Connections
```

-------------------------

> .. image:: images/preemptive-troubleshooting/preCheckBatt.png

Reviewer 2: It would be absolutely amazing if we could just... replace this picture.

Author: Does the image I added in 5100856 work well?

Apply considerable force onto the battery cable at 90 degrees to try to move the direction of the cable leaving the battery, if successful the connection was not tight enough to begin with and it should be redone.

> Secure the battery to robot connection

Reviewer 2: ```suggestion
Securing the Battery to the Robot
```

--------------------------------------

.. image:: images/preemptive-troubleshooting/preCheckConnecc.png

In almost every event we see at least one robot where a not properly secured battery connector (the large Anderson) comes apart and disconnects power from the robot. This has happened in championship matches on the Einstein and everywhere else. Its an easy to ensure that this doesn't happen to you by securing the two connectors by wrapping a tie wrap around the connection. 10 or 12 tie wraps for the piece of mind during an event is not a high price to pay to guarantee that you will not have the problem of this robot from an actual event after a bumpy ride over a defense.

> 120 Amp circuit breaker

Reviewer 2: ```suggestion
120 Amp Circuit Breaker
```

-----------------------

> .. image:: images/preemptive-troubleshooting/preCheckBreaker.png

Reviewer 2: If we could also somehow replace this image also...

Author: Does the image I added in 5100856 work well?

Apply a twisting force onto the cable to rotate the harness.  If you are successful then the screw is not tight enough.  Split washers might help here, but in the mean time, these require checking every few matches.

Because the metal is just molded into the case, every once in awhile you will break off the bolt, ask any veteran team and they’ll tell you they go through a number of these every few seasons.  After tightening the nut, retest by once again trying to twist the cable.

Power Distribution Panel (PDP)
------------------------------

> .. image:: images/preemptive-troubleshooting/preCheckPDP.png

Reviewer 2: This image should probably be replaced as well.

Author: Does the image I added in 5100856 work well?

Make sure that split washers were placed under the PDP screws, but it is not easy to visually confirm, and sometimes you can’t.  You can check by removing the case.    Also if you squeeze the red and black wires together, sometimes you can catch the really lose connections.

> Tug test everything

Reviewer 2: ```suggestion
Tug Testing
```

-------------------

> .. image:: images/preemptive-troubleshooting/preCheckTug.png

Reviewer 2: This should be replaced as well.

Author: Does the image I added in 5100856 work well?

Spade connectors can also fail due to improper crimps, so tug-test those as well.

> Blade fuses

Reviewer 2: ```suggestion
Blade Fuses
```

-----------

.. image:: images/preemptive-troubleshooting/preCheckFuse.png

If you can remove the blade fuses by hand then they are not in completely. Make sure that they are completely seated in the PDP so that they don't pop out during robot operation.

roboRIO swarf
-------------

Swarf is: fine chips or filings of stone, metal, or other material produced by a machining operation. Often modifications must be made to a robot while the control system parts are in place. The circuit board for the roboRIO is conformally coated, but that doesn't absolutely guarantee that metal chips won't short out traces or components inside the case. In this case, you must exercise care in making sure that none of the chips end up in the roboRIO or any of the other components. In particular, the exposed 3 pin headers are a place where chips can enter the case. A quick sweep through each of the four sides with a flashlight is usually sufficient to find the really bad areas of infiltration.

> Radio barrel jack

Reviewer 2: ```suggestion
Radio Barrel Jack
```

-----------------

Make sure the correct barrel jack is used, not one that is too small and falls out for no reason.  This isn’t common, but ask an FTA and every once in awhile a team will use some random barrel jack that is not sized correctly, and it falls out in a match on first contact.

> Ethernet cable

Reviewer 2: ```suggestion
Ethernet Cable
```

--------------

If the RIO to radio ethernet cable is missing the clip that locks the connector in, get another cable.  This is a common problem that will happen several times in every competition. Make sure that your cables are secure. The clip often breaks off, especially when pulling it through a tight path, it snags on something then breaks.

> Cable slack

Reviewer 2: ```suggestion
Loose Cables
```

-----------

Cables must be tightened down, particularly the radio power and ethernet cable.  The radio power cables don’t have a lot of friction force and will fall out (even if it is the correct barrel) if the weight of the cable-slack is allowed to swing freely.

Ethernet cable is also pretty heavy, if it’s allowed to swing freely, the plastic clip may not be enough to hold the ethernet pin connectors in circuit.

> Reproducing problems in the pit

Reviewer 2: ```suggestion
Reproducing Problems in the Pit
```

Robot inspectors do this, but you should do it as well, it helps robot inspectors out and they appreciate it.  And it guarantees that you are running with the most recent, bug fixed code. You wouldn't want to lose a match because of an out of date piece of control system software on your robot.

> Driver station checks

Reviewer 2: ```suggestion
Driver Station Checks
```

- Verify that there is nothing unnecessary running in the application tray in the start menu (bottom right side)

> Handy tools

Reviewer 2: ```suggestion
Handy Tools
```

-----------

> .. image:: images/preemptive-troubleshooting/preCheckTools.png

Reviewer 2: If this could be replaced with a non-blown out picture as well..

Author: Does the image I added in 5100856 work well?
